storage/innobase/handler/handler0alter.cc

```diff
@@ -866,11 +871,10 @@ ha_innobase::check_if_supported_inplace_alter(
 
 	const bool need_rebuild = innobase_need_rebuild(ha_alter_info, table);
 
-	if (need_rebuild
-	    && (table->versioned(VERS_TIMESTAMP)
-		|| altered_table->versioned(VERS_TIMESTAMP))) {
+	if (altered_table->versioned(VERS_TIMESTAMP)
+	    && (ha_alter_info->handler_flags & INNOBASE_ALTER_VERSIONED_REBUILD)) {
```

Comment:
I would keep the `need_rebuild &&` condition. I see no reason to block (for example) changes of column comments or default values. Tests passed for me when I restored that condition.